git_jira/git_jira.py

```diff
@@ -32,28 +39,42 @@ def load_branches():
 
     for issue in response['issues']:
         formatted = issue['key'] + " " + issue['fields']['summary']
-        formatted_branches.append(re.sub(r"[^a-zA-Z0-9]+", ' ', formatted))
+        formatted_summary = re.sub(r"[^a-zA-Z0-9]+", ' ', formatted)
+        formatted_branches.append({
+            'summary': formatted_summary,
+            'issuetype': issue['fields']['issuetype']
+        })
     return formatted_branches[:MAX_RESULT]
 
-
 def main(prefix: Annotated[str, typer.Option(help="Prefix that is being used for the new branch.")] = "feature",
-         no_prefix: Annotated[bool, typer.Option("--no-prefix", help="Will not use a prefix")] = False):
+         no_prefix: Annotated[bool, typer.Option("--no-prefix", help="Will not use a prefix")] = False,
+         auto_branch_prefix: Annotated[bool, typer.Option("--auto-branch-prefix", help="Automatically determine branch prefix based on ticket type")] = False):
```

Comment:
If you do https://github.com/freenowtech/git-jira/pull/2/files#r1477907772, I think you can drop this (=> only map if the variable is defined).